appear/src/main/java/com/arcbees/gquery/appear/client/Appear.java

```diff
@@ -0,0 +1,218 @@
+import com.google.gwt.query.client.Function;
+import com.google.gwt.query.client.GQuery;
+import com.google.gwt.query.client.plugin.Observe;
+import com.google.gwt.query.client.plugins.Plugin;
+import com.google.gwt.user.client.Window;
+
+public class Appear extends GQuery {
+    public static final Class<Appear> Appear = GQuery.registerPlugin(Appear.class, new Plugin<Appear>() {
+        public Appear init(GQuery gq) {
+            return new Appear(gq);
+        }
+    });
+
+    private static final String DISAPPEAR_EVENT = "disappear";
+    private static final String APPEAR_EVENT = "appear";
+    private static final Map<Element, AppearImpl> impls = new HashMap<Element, AppearImpl>();
+
+    private static HandlerRegistration handlerRegistration;
+
+    protected Appear(GQuery gq) {
+        super(gq);
+    }
+
+    public Appear appear(Function... f) {
+        on(APPEAR_EVENT, f);
+
+        for (Element element : elements()) {
+            AppearImpl impl = getOrCreateImpl(element);
+            impl.setAppear(true);
+        }
+
+        maybeInitHandlers();
+
+        return this;
+    }
+
+    public Appear appearOff() {
+        off(APPEAR_EVENT);
+
+        for (Element element : elements()) {
+            AppearImpl impl = getImpl(element);
+            if (impl != null) {
+                if (impl.isDisappear()) {
+                    impl.setAppear(false);
+                } else {
+                    impls.remove(impl);
+                }
+            }
+        }
+
+        maybeUnbindHandlers();
+
+        return this;
+    }
+
+    public Appear disappear(Function... f) {
+        on(DISAPPEAR_EVENT, f);
+
+        for (Element element : elements()) {
+            AppearImpl impl = getOrCreateImpl(element);
+
+            impl.setDisappear(true);
+        }
+
+        maybeInitHandlers();
+
+        return this;
+    }
+
+    public Appear disappearOff() {
+        off(DISAPPEAR_EVENT);
+
+        for (Element element : elements()) {
+            AppearImpl impl = getImpl(element);
+
+            if (impl != null) {
+                if (impl.isAppear()) {
+                    impl.setDisappear(false);
+                } else {
+                    impls.remove(impl);
+                }
+            }
+        }
+
+        maybeUnbindHandlers();
+
+        return this;
+    }
+
+    private void maybeUnbindHandlers() {
+        if (impls.isEmpty()) {
+            handlerRegistration.removeHandler();
+            handlerRegistration = null;
+        }
+    }
+
+    private void maybeInitHandlers() {
+        if (handlerRegistration == null) {
+            final HandlerRegistration scrollHandler = Window.addWindowScrollHandler(new Window.ScrollHandler() {
+                @Override
+                public void onWindowScroll(Window.ScrollEvent event) {
+                    checkElementsAppearance();
+                }
+            });
+
+            final Observe observeHandler = $(Document.get()).as(Observe.Observe)
```

Comment:
It would be nice to make that optionnal (at least be able to desactivate that)
I would see also a public method that allow the developper to trigger manually the checkElementsAppearance() method